tinygrad/tensor.py

```diff
@@ -379,8 +379,8 @@ def _reduce(self, fxn:Type[Function], axis:Optional[Union[int, Tuple[int, ...]]]
     return ret if keepdim else ret.reshape(shape=shape)
 
   def sum(self, axis=None, keepdim=False): return self._reduce(mlops.Sum, axis, keepdim)
-  def max(self, axis=None, keepdim=False): return self._reduce(mlops.Max, axis, keepdim)
   def min(self, axis=None, keepdim=False): return -((-self).max(axis=axis, keepdim=keepdim))
+  def max(self, axis=None, keepdim=False, ignore_nan=True): return self._reduce(mlops.Max, axis, keepdim) if ignore_nan else self._reduce(mlops.Max, axis, keepdim) + (self.isnan() * np.nan).sum()
```

Comment:
https://pytorch.org/docs/stable/generated/torch.fmax.html

Reply:
The correct function isn't actually `fmax` (`fmax` is the nan ignoring version of elementwise `maximum`), it should be `nanmax` but that's not implemented in torch yet: https://github.com/pytorch/pytorch/issues/61474